specs/exec-engine.md

```diff
@@ -132,9 +137,43 @@ can be accessed in two interchangeable ways:
     - Overhead as big-endian `uint256` in slot `5`
     - Scalar as big-endian `uint256` in slot `6`
 
-#### Eclipse L1-Cost fee changes (EIP-4844 DA)
+#### Ecotone L1-Cost fee changes (EIP-4844 DA)
+
+Ecotone allows posting batches via Blobs which are subject to a new fee market. To account for this feature,
+L1 cost is computed as:
+
+`(compressedTxSize) * (l1Basefee*16*lBasefeeScalar + l1BlobBasefeeScalar*l1BlobBasefeeScalar) / 1e6`
+
+Where:
+
+- the computation is an unlimited precision integer computation, with the result in Wei and having
+  `uint256` range.
 
-TBD.
+- `compressedTxSize` is an approximation of how many bytes the transaction occupies in a compressed
+  batch. It is determined from the *full* encoded transaction as: `compressedTxSize = (zeroes*4 +
+  ones*16) / 16` (To preserve precision under integer arithmetic, the division by 16 is actually
+  performed at the very end of the fee computation together with the division by 1e6 as a single
+  division by 16e6.)
+
+- `l1Basefee` is the L1 basefee of the latest L1 origin registered in the L2 chain.
+
+- `l1BlobBasefee` is the blob gasprice, computed as described in [EIP-4844][4844-gas] from the
+  header of the latest registered L1 origin block.
+
+[4844-gas]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4844.md#gas-accounting
+
+The two basefee values and their respective scalars can be accessed in two interchangeable
+ways:
+
+- read from the deposited L1 attributes (`l1BasefeeScalar`, `l1BlobBasefeeScalar`, `basefee`,
+  `blobBasefee`) of the current L2 block
+- read from the L1 Block Info contract (`0x4200000000000000000000000000000000000015`)
+  - using the respective solidity getter functions
+  - using direct storage-reads:
+    - basefee as big-endian `uint256` in slot `1`
+    - blobBasefee as big-endian `uint256` in slot `7`
+    - l1BasefeeScalar as big-endian `uint32` in first 4 bytes of slot `8`
```

Comment:
There is an inconsistency here - what does first 4 bytes mean? Relative to the msb or lsb? Big endian or little endian? `solc` prefers the "offset" language. The [solidity docs](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html) don't have any good examples. To get the offset, you can use the following command:

```
forge inspect SystemConfig storageLayout
```

Reply:
I don't think you need to specify msb or lsb or endianness if you use the same offset terminology that solc uses. iirc the offset is relative to the lsb

Reply:
Yup, now that I finished the contract change I understand this better.  Updated to use slot+offset.